components/clickup/clickup.app.js

```diff
@@ -0,0 +1,205 @@
+const axios = require("axios");
+
+module.exports = {
+  type: "app",
+  app: "clickup",
+  propDefinitions: {
+    workspace: {
+      type: "string",
+      label: "Workspace",
+      async options() {
+        const workspaces = (await this.getWorkspaces()).teams;
+        return workspaces.map((workspace) => {
+          return {
+            label: workspace.name,
+            value: workspace.id,
+          };
+        });
+      },
+    },
+    space: {
+      type: "string",
+      label: "Space",
+      async options({ workspace }) {
+        const spaces = (await this.getSpaces(workspace)).spaces;
```

Comment:
@dylburger isn't it possible to access `this.workspace` here if that prop is provided, rather than explicitly passing it in with a function? If so, is there a preference?

Reply:
The way we implemented the `options` method, the values of props that have been configured at runtime, and which future props' `options` methods need access to, must be passed in the current manner, e.g.

```javascript
propDefinition: [
  common.props.clickup,
  "status",
  (c) => ({
    list: c.list,
  }),
],
```

I am forgetting the core _reason_ we implemented it this way (so I asked the implementor). 

I realized we also don't have docs in this in the component API, so I added an issue to write those.